util/verible-format.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+import argparse
+import subprocess
+import logging
+import difflib
+
+logger = logging.getLogger('verible_format')
+
+VERIBLE_ARGS = ["--formal_parameters_indentation=indent",
+                "--named_parameter_indentation=indent",
+                "--named_port_indentation=indent",
+                "--port_declarations_indentation=indent"]
+
+
+def get_repo_top():
+    repo_top = subprocess.Popen('git rev-parse --show-toplevel',
+                                stdout=subprocess.PIPE, shell=True)
+    return repo_top.stdout.read().decode().strip()
+
+
+def get_verible_executable_path():
+    proc = subprocess.Popen('which verible-verilog-format',
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=True)
+    return proc.stdout.read().decode().strip()
```

Comment:
I'd suggest turning this function into something much more like `which`, so it returns the path if found and None otherwise. Maybe something like this?
```python
def get_verible_executable_path():
    try:
        return subprocess.run(['which', 'env'],
                              check=True, text=True,
                              capture_output=True).stdout.strip()
    except subprocess.CalledProcessError:
        return None
```
If you do that then `is_verible_formatter_installed` isn't needed. Indeed, you can call this at the start and then just pass the resulting path around. So functions like `get_verible_version` would then take a `verible_path` argument.

Reply:
Makes sense. Using `verible_exec_path = get_verible_executable_path()`.